content/blog/2019/07/2019-07-04-cloudfeature-gsoc-cp1.adoc

```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+title: "GSoC Project Progress : Cloud features for external workspace manager"
+tags:
+- jenkins
+- gsoc
+author: yufei_zhang
+description: "Short summary of coding period 1" # optional
+opengraph:
+---
+
+=== About Me
+
+My name is Yufei Zhang, a student from Zhejiang University and Simon Fraser University Dual Degree Program. I want to share the discoveries when we are exploring the possiblities of extending current link:https://plugins.jenkins.io/external-workspace-manager[external workspace manager plugin] (referred to as EWM) to support AWS EFS. 
+
+=== Backgrounds
+
+For users, they might want to provision a high-available Jenkins cluster, together with other services such as Keycloak for SSO, and use k8s to autoscale agents. All these operations needs a lot of configuration and cloud resources if they decide to base Jenkins on clouds. 
+
+In such cases, EWM was designed to give multiple agents access to shared disks, hence reduce the cost of copying duplicated data, such as codes and outputs in the workspace, or archives that needs to be managed by other services (e.g. Some companies have a ec2 instance running a cron job, pulling the Jenkins build artifacts and store in it S3). With these backgrounds, it is natural to think of "supporting cloud disks".
+
+Our project proposal was based on extending cloud features, but then we discovered we were on the wrong track.
+
+=== Discoveries
+
+In order to make things clear, we need to understand the true nature of EWM. As we stated earlier, we want multiple agents to share disk spaces, and the main obstacle was they all have their own workspace directory (paths) by default.
+If we can modify the default workspace directory, then we might be able to utilize the shared NFS (Network File System).
+And that is what EWM did.
+There are other features in like different strategies choosing a disk, but the way EWM to solve the problem is simple but effective.
+
+And by the way, from my perspective,the main problem, synchronization when visiting shared resources like two slave visit the shared file, is guaranteed by the implementation of the NFS itself, and the EWM didn't offer any synchronization mechanism to support concurrent execution.
+Jenkins itself is not designed to be synchronized.
+It is left to the users to partition their jobs and let the user manage syncronization.
```

Comment:
It is left to the users to partition their jobs and let the user manage synchronization.

Reply:
fixed minor spelling typo:
syncronization -> synchronization

Reply:
@martinda I don’t have permission in your branch (according to GitHub) to commit the minor typo fix suggestion.  Thus, you will have to commit my suggestion to fix the minor typo.  Thank you.

Reply:
I assigned this PR to @martinda to commit:
```
fixed minor spelling typo:
syncronization -> synchronization
```